contracts/identity/Identity.sol

```diff
@@ -35,47 +38,61 @@ contract Identity is ProxyStorage {
         address payable to,
         uint256 value,
         bytes memory data,
-        uint64 fees,
+        uint256 baseFee,
+        uint256 gasPrice,
+        uint256 gasLimit,
         address currencyNetworkOfFees,
         uint256 nonce,
+        uint256 timeLimit,
+        uint8 operationType,
         bytes memory extraData,
         bytes memory signature
     )
-        public returns (bool _success)
+        public
     {
-
+        uint startGas = gasleft();
         require(from == address(this), "The transaction is not meant for this identity contract");
 
         bytes32 hash = transactionHash(
             from,
             to,
             value,
             data,
-            fees,
+            baseFee,
+            gasPrice,
+            gasLimit,
             currencyNetworkOfFees,
             nonce,
+            timeLimit,
+            operationType,
             extraData
```

Comment:
the original creators even removed that field themselves, maybe we should also think of removing it.